### ISSUE TITLE
Variable inferred as unbound although defined

### ISSUE THREAD

**Issue by syagev** [microsoft/pylance-release#2123]
## Environment data

-   Language Server version: Pylance language server 2021.11.2 (pyright b27108d5)
-   OS and version: Win 10.0.19044
-   Python version (& distribution if applicable, e.g. Anaconda): CPython 3.6

## Expected behaviour

The following code should be valid:
```python
new_mount_path = Path(tempfile.gettempdir()) / f'flsync_union_mnt-{mount_uid}'
drive_letter = '*' if os.name == 'nt' else str(new_mount_path)
```

## Actual behaviour

`new_mount_path` generates error 

> Argument type is unknown
> Argument corresponds to parameter "`object`" in function "`__new__`" 
> `PylancereportUnknownArgumentType`

Additionally, hovering over the parameter we see it's value is unbound:
![image](https://user-images.githubusercontent.com/5247438/144116018-df0baf22-e47b-42a8-99e0-eed63246c618.png)

## Logs

<details>

<summary>Logs</summary>

```

[Info  - 21:35:11] Pylance language server 2021.11.2 (pyright b27108d5) starting
[Info  - 21:35:11] Server root directory: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist
[Info  - 21:35:11] Setting pythonPath for service "python": "C:\Users\styagev\.virtualenvs\flsync\Scripts\python.exe"
[Info  - 21:35:11] Loading configuration file at d:\Projects\flsync\flsync\python\pyrightconfig.json
[Info  - 21:35:12] Assuming Python version 3.6
[Info  - 21:35:12] Assuming Python platform Windows
[Info  - 21:35:12] No include entries specified; assuming d:\Projects\flsync\flsync\python
Search paths for d:\Projects\flsync\flsync\python
  c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib
  d:\Projects\flsync\flsync\python
  d:\Projects\flsync\flsync\python\src
  d:\Projects\flsync\flsync\python\typings
  c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stubs\...
  c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\bundled\stubs
  c:\Program Files\Python36\DLLs
  c:\Program Files\Python36\Lib
  c:\Program Files\Python36
  c:\Users\styagev\.virtualenvs\flsync
  c:\Users\styagev\.virtualenvs\flsync\Lib\site-packages
  d:\Projects\flsync\flsync\python\src
[Info  - 21:35:12] Searching for source files
[Info  - 21:35:12] Found 11 source files
[Info  - 21:35:12] Background analysis(1) root directory: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist
[Info  - 21:35:12] Background analysis(1) started
Background analysis message: setConfigOptions
Background analysis message: setImportResolver
Background analysis message: ensurePartialStubPackages
Background analysis message: setTrackedFiles
Background analysis message: markAllFilesDirty
Background analysis message: analyze
Background analysis message: markFilesDirty
Background analysis message: analyze
Background analysis message: invalidateAndForceReanalysis
[Info  - 21:35:18] Searching for source files
[Info  - 21:35:18] Found 11 source files
Background analysis message: setTrackedFiles
Background analysis message: markAllFilesDirty
Background analysis message: analyze
Background analysis message: markFilesDirty
Background analysis message: analyze
Background analysis message: invalidateAndForceReanalysis
Background analysis message: invalidateAndForceReanalysis
Background analysis message: markFilesDirty
Background analysis message: markFilesDirty
Background analysis message: markFilesDirty
[Info  - 21:35:20] Searching for source files
[Info  - 21:35:20] Found 11 source files
Background analysis message: setTrackedFiles
Background analysis message: markAllFilesDirty
Background analysis message: analyze
Background analysis message: setFileOpened
[FG] parsing: d:\Projects\flsync\flsync\python\src\flsync\rclone.py (203ms)
[FG] parsing: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\builtins.pyi [fs read 13ms] (232ms)
[FG] binding: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\builtins.pyi (95ms)
[FG] binding: d:\Projects\flsync\flsync\python\src\flsync\rclone.py (6ms)
Background analysis message: getDiagnosticsForRange
Background analysis message: analyze
[BG(1)] analyzing: d:\Projects\flsync\flsync\python\src\flsync\rclone.py ...
[BG(1)]   parsing: d:\Projects\flsync\flsync\python\src\flsync\rclone.py (130ms)
[BG(1)]   parsing: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\builtins.pyi [fs read 7ms] (227ms)
[BG(1)]   binding: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\builtins.pyi (97ms)
[BG(1)]   binding: d:\Projects\flsync\flsync\python\src\flsync\rclone.py (7ms)
[BG(1)]   checking: d:\Projects\flsync\flsync\python\src\flsync\rclone.py ...
[BG(1)]     parsing: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\typing.pyi [fs read 1ms] (53ms)
[BG(1)]     binding: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\typing.pyi (31ms)
[BG(1)]     parsing: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\_typeshed\__init__.pyi [fs read 1ms] (24ms)
[BG(1)]     binding: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\_typeshed\__init__.pyi (3ms)
[BG(1)]     parsing: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\typing_extensions.pyi [fs read 0ms] (4ms)
[BG(1)]     binding: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\typing_extensions.pyi (6ms)
[BG(1)]     parsing: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\abc.pyi [fs read 0ms] (5ms)
[BG(1)]     binding: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\abc.pyi (1ms)
[BG(1)]     parsing: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\logging\__init__.pyi [fs read 0ms] (35ms)
[BG(1)]     binding: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\logging\__init__.pyi (13ms)
[BG(1)]     parsing: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\os\__init__.pyi [fs read 0ms] (26ms)
[BG(1)]     binding: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\os\__init__.pyi (26ms)
[BG(1)]     parsing: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\pathlib.pyi [fs read 1ms] (14ms)
[BG(1)]     binding: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\pathlib.pyi (8ms)
[BG(1)]     parsing: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\re.pyi [fs read 1ms] (8ms)
[BG(1)]     binding: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\re.pyi (6ms)
[BG(1)]     parsing: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\subprocess.pyi [fs read 1ms] (14ms)
[BG(1)]     binding: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\subprocess.pyi (11ms)
[BG(1)]     parsing: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\tempfile.pyi [fs read 1ms] (9ms)
[BG(1)]     binding: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\tempfile.pyi (9ms)
[BG(1)]     parsing: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\urllib\parse.pyi [fs read 1ms] (4ms)
[BG(1)]     binding: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\urllib\parse.pyi (4ms)
[BG(1)]     parsing: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\uuid.pyi [fs read 1ms] (9ms)
[BG(1)]     binding: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\uuid.pyi (2ms)
[BG(1)]     parsing: d:\Projects\flsync\flsync\python\src\flsync\path_mapper.py [fs read 0ms] (79ms)
[BG(1)]     binding: d:\Projects\flsync\flsync\python\src\flsync\path_mapper.py (2ms)
[BG(1)]     parsing: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\enum.pyi [fs read 1ms] (5ms)
[BG(1)]     binding: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\enum.pyi (9ms)
[BG(1)]     parsing: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\types.pyi [fs read 0ms] (35ms)
[BG(1)]     binding: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\types.pyi (10ms)
[BG(1)]     parsing: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\collections\abc.pyi [fs read 1ms] (2ms)
[BG(1)]     binding: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\collections\abc.pyi ...
[BG(1)]       parsing: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\_collections_abc.pyi [fs read 1ms] (3ms)
[BG(1)]       binding: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\_collections_abc.pyi (1ms)
[BG(1)]     binding: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\collections\abc.pyi (4ms)
[BG(1)]   checking: d:\Projects\flsync\flsync\python\src\flsync\rclone.py (985ms)
[BG(1)] analyzing: d:\Projects\flsync\flsync\python\src\flsync\rclone.py (1450ms)
Background analysis message: getDiagnosticsForRange
Background analysis message: resumeAnalysis
Background analysis message: getDiagnosticsForRange
Background analysis message: getDiagnosticsForRange
Background analysis message: getDiagnosticsForRange
Background analysis message: getDiagnosticsForRange
Background analysis message: getDiagnosticsForRange
Background analysis message: getDiagnosticsForRange
Background analysis message: setFileOpened
Background analysis message: markFilesDirty
Background analysis message: analyze
[BG(1)] analyzing: d:\Projects\flsync\flsync\python\src\flsync\rclone.py ...
[BG(1)]   parsing: d:\Projects\flsync\flsync\python\src\flsync\rclone.py (16ms)
[BG(1)]   binding: d:\Projects\flsync\flsync\python\src\flsync\rclone.py (3ms)
[BG(1)]   checking: d:\Projects\flsync\flsync\python\src\flsync\rclone.py (167ms)
[BG(1)] analyzing: d:\Projects\flsync\flsync\python\src\flsync\rclone.py (187ms)
Background analysis message: getDiagnosticsForRange
Background analysis message: resumeAnalysis
Background analysis message: getDiagnosticsForRange
[FG] parsing: d:\Projects\flsync\flsync\python\src\flsync\rclone.py (14ms)
[FG] binding: d:\Projects\flsync\flsync\python\src\flsync\rclone.py (1ms)
Background analysis message: getDiagnosticsForRange
Background analysis message: getDiagnosticsForRange
Background analysis message: getDiagnosticsForRange
Background analysis message: getDiagnosticsForRange
[FG] parsing: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\typing.pyi [fs read 0ms] (34ms)
[FG] binding: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\typing.pyi (13ms)
[FG] parsing: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\_typeshed\__init__.pyi [fs read 1ms] (13ms)
[FG] binding: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\_typeshed\__init__.pyi (1ms)
[FG] parsing: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\typing_extensions.pyi [fs read 0ms] (2ms)
[FG] binding: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\typing_extensions.pyi (1ms)
[FG] parsing: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\abc.pyi [fs read 0ms] (1ms)
[FG] binding: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\abc.pyi (0ms)
[FG] parsing: d:\Projects\flsync\flsync\python\src\flsync\path_mapper.py [fs read 0ms] (3ms)
[FG] binding: d:\Projects\flsync\flsync\python\src\flsync\path_mapper.py (1ms)
[FG] parsing: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\pathlib.pyi [fs read 0ms] (8ms)
[FG] binding: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\pathlib.pyi (3ms)
[FG] parsing: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\os\__init__.pyi [fs read 1ms] (22ms)
[FG] binding: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\os\__init__.pyi (15ms)
[FG] parsing: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\logging\__init__.pyi [fs read 0ms] (21ms)
[FG] binding: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\logging\__init__.pyi (8ms)
Background analysis message: getDiagnosticsForRange
Background analysis message: getDiagnosticsForRange
[FG] parsing: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\tempfile.pyi [fs read 1ms] (7ms)
[FG] binding: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\tempfile.pyi (2ms)
[FG] parsing: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\subprocess.pyi [fs read 1ms] (6ms)
[FG] binding: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\subprocess.pyi (7ms)
[FG] parsing: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\uuid.pyi [fs read 1ms] (4ms)
[FG] binding: c:\Users\styagev\.vscode\extensions\ms-python.vscode-pylance-2021.11.2\dist\typeshed-fallback\stdlib\uuid.pyi (1ms)
Background analysis message: getDiagnosticsForRange
Background analysis message: getDiagnosticsForRange
Background analysis message: getDiagnosticsForRange
Background analysis message: getDiagnosticsForRange
Background analysis message: getDiagnosticsForRange
Background analysis message: getDiagnosticsForRange
```

</details>

## Code Snippet / Additional information

IMO it has something to do with the `os.name == 'nt'` conditional, since changing it to `os.name != 'nt'` makes the error go away. Also, testing the value of that parameter in runtime produces a results as expected:
```python
>>> new_mount_path
WindowsPath('C:/Users/styagev/AppData/Local/Temp/flsync_union_mnt-46d727')
```

I was unable to easily create a minimal repro, so I'm attaching the entire file:

<details>
<summary>rclone.py</summary>

```python
""" rclone wrappers """

import logging
import os
from pathlib import Path
import re
import subprocess
import tempfile
from urllib.parse import urlparse
import uuid
from typing import IO, Dict, Match, Optional, cast

from .path_mapper import PeerDef


_LOG = logging.getLogger('flsync')


class _RCloneState():
    def __init__(self) -> None:
        self.mounts = {}                # type: Dict[Path, PeerDef]
        self.mount_path = None          # type: Optional[Path]
        self.remote_mounts_path = None  # type: Optional[Path]

        # the rclone proc that mounts a union of the remotes (with VFS enabled)
        self.remote_union_proc = None   # type: Optional[subprocess.Popen[str]]

        # the rclone proc that mounts a union of the above union together with the local path (without VFS)
        self.union_proc = None          # type: Optional[subprocess.Popen[str]]

    def is_peer_mounted(self, root_path: Path) -> bool:
        """ Checks whether the peer is already mounted """
        return root_path in self.mounts

_RC = _RCloneState()


def mount_peer(peer_def: PeerDef, local_path: Path) -> Path:
    """ Mount the specified peer (if required) """

    # normalize the peer's root path so that D:\Datasets matches D:/Datasets and D:/datasets etc.
    peer_root_path = Path(peer_def['root-path'])

    # if the peer is mounted, return the existing mount path
    if not _RC.is_peer_mounted(peer_root_path):

        mount_uid = str(uuid.uuid4())[-6:]  # UID for the mount to avoid collisions in the temp filesystem

        def _verify_mount_succeeded(rclone_proc: 'subprocess.Popen[str]') -> Optional[Path]:
            """ utility function listening on stdout for rclone's confirmation that mounting succeeded """

            drive_letter = None # type: Optional[Match[str]]
            for out_line in iter(cast(IO[str], rclone_proc.stderr).readline, ''):
                if drive_letter is None:
                    drive_letter = re.search('Assigning drive letter "([A-Za-z]:)"', out_line)
                if re.match('The service .* has been started', out_line) is not None:
                    return None if drive_letter is None else Path(drive_letter.group(1))

            raise RCloneException('Failed to mount remote union')


        if peer_def['type'] != 'local':
            # if the new peer is a remote, re-mount a union of the remotes
            _RC.remote_mounts_path = Path(tempfile.gettempdir()) / f'flsync_remote_mnt-{mount_uid}'
            remote_mount_vfs_cache_pathname = str(_RC.remote_mounts_path) + '-vfs'

            remote_union_args = {
                'RCLONE_UNION_UPSTREAMS': ' '.join(
                    [
                        f'"{_connection_string(mounted_peer_def)}"'
                        for mounted_peer_def in filter(
                            lambda mounted_peer_def: mounted_peer_def['type'] != 'local',
                            _RC.mounts.values())
                    ] +
                    [f'"{_connection_string(peer_def)}"']
                ),
                'RCLONE_UNION_SEARCH_POLICY': 'epall',
                'RCLONE_UNION_ACTION_POLICY': 'epff',
                'RCLONE_UNION_CREATE_POLICY': 'epff'
            }

            if _RC.remote_union_proc is not None:
                _RC.remote_union_proc.terminate()
            _RC.remote_union_proc = subprocess.Popen(
                [
                    'rclone', 'mount', ':union:',
                    str(_RC.remote_mounts_path),
                    '--cache-dir', remote_mount_vfs_cache_pathname,
                    '--vfs-cache-mode', 'full',
                    '--vfs-cache-max-age', '99h',
                    '--read-only'
                ],
                env=remote_union_args,
                universal_newlines=True,
                stdout=subprocess.PIPE,
                stderr=subprocess.PIPE)

            _verify_mount_succeeded(_RC.remote_union_proc)

        _RC.mounts[peer_root_path] = peer_def

        # the primary mount - a union of the previous union and the local paths
        new_mount_path = Path(tempfile.gettempdir()) / f'flsync_union_mnt-{mount_uid}'

        drive_letter = '*' if os.name == 'nt' else str(new_mount_path)

        union_args = {
            'RCLONE_UNION_UPSTREAMS': ' '.join(
                [f'"{local_path}"'] +
                [f'"{local_peer_def["root-path"]}"' for local_peer_def in
                 filter(lambda local_peer_def: local_peer_def['type'] == 'local', _RC.mounts.values())] +
                ([f'"{_RC.remote_mounts_path}"'] if _RC.remote_mounts_path is not None else [])
            ),
            'RCLONE_UNION_SEARCH_POLICY': 'epall',
            'RCLONE_UNION_ACTION_POLICY': 'epff',
            'RCLONE_UNION_CREATE_POLICY': 'epff'
        }

        if _RC.union_proc is not None:
            _RC.union_proc.terminate()

        _RC.union_proc = subprocess.Popen(
            [
                'rclone', 'mount', ':union:',

            ],
            env=union_args,
            universal_newlines=True,
            stdout=subprocess.PIPE,
            stderr=subprocess.PIPE)

        win_net_drive_path = _verify_mount_succeeded(_RC.union_proc)
        _RC.mount_path = win_net_drive_path if os.name == 'nt' else new_mount_path


        _LOG.info('Mounted "%s"', peer_def['root-path'])

    return cast(Path, _RC.mount_path)


class RCloneException(Exception):
    """ Represents an error from rclone """
    pass


def _connection_string(peer_def: PeerDef) -> str:
    try:
        if peer_def['type'] == 'azureblob':
            try:
                # currently we only support container URLs (not nested folders)
                container_name = urlparse(peer_def['url']).path
                if '/' in container_name:
                    raise ValueError('Azure Blob peer must be defined as the entire container, nested folder '
                                     f'"{container_name}" not allowed')

                return (
                    ":azureblob,"
                    f"sas_url=''{peer_def['url']}{peer_def['auth']['sas-query-str']}'':{container_name},"
                    "read_only"
                    )

            except KeyError as ex:
                raise ValueError(f'Malformed azureblob peer definition\n{peer_def}') from ex

        raise ValueError(f'Unknown peer type "{peer_def["type"]}"')
    except KeyError as ex:
        raise ValueError(f'Malformed peer definition\n{peer_def}') from ex
```

</details>

**Comment by erictraut** [comment]
Thanks for the bug report. Here's a minimal, self-contained repro. I'll investigate further.

```python
# pyright: strict

import os

x = 1
y = "" if os.name == "nt" else str(x)
```

**Comment by syagev** [comment]
Ahhh I had that in my editor in a new file, in my active multi-folder workspace (configured to 'strict' at the **folder** level), and the error didn't show up! 

Apparently VSCode applies the 'strict' mode only after it's saved in the specific folder (the other folder I have in the workspace does not have 'strict' mode).

I guess this behavior makes sense but it sure drove me a little crazy!

Anyway thanks for ack :)

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by bschnurr** [comment]
This issue has been fixed in version 2021.12.0, which we've just released. You can find the changelog here:  [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2021120-2-december-2021)
